### PR TITLE
Add A Better I18n Support

### PR DESCRIPTION
**Pull request by TartaricAcid**

### CODE REVIEW

**Comment by mc-kaishixiaxue** [comment]
@TartaricAcid I found a typo in zh_cn.lang:
Line 5:`item.theoneprobe.probe_goggles.name=检测器(视频)`
The right name is "检测器（饰品）",don't forget to correct it.

File: src/main/java/mcjty/theoneprobe/apiimpl/providers/DefaultProbeInfoProvider.java [hunk 1]
```diff
@@ -144,24 +141,24 @@ private void showRedstonePower(IProbeInfo probeInfo, World world, IBlockState bl
...
+                    .text(I18n.format("gui.theoneprobe.default_provider.rf.power", LABEL, INFO, redstonePower));
         }
     }
 
     private void showLeverSetting(IProbeInfo probeInfo, World world, IBlockState blockState, IProbeHitData data, Block block) {
         if (block instanceof BlockLever) {
             Boolean powered = blockState.getValue(BlockLever.POWERED);
             probeInfo.horizontal().item(new ItemStack(Items.REDSTONE), probeInfo.defaultItemStyle().width(14).height(14))
-                    .text(LABEL + "State: " + INFO + (powered ? "On" : "Off"));
+                    .text(I18n.format("gui.theoneprobe.default_provider.lever.state", LABEL, INFO, (powered ? "On" : "Off")));
```
Comment (Snownee) [inline src/main/java/mcjty/theoneprobe/apiimpl/providers/DefaultProbeInfoProvider.java:152]: I18n for "On" & "Off"?

File: src/main/java/mcjty/theoneprobe/apiimpl/providers/DefaultProbeInfoProvider.java [hunk 2]
```diff
@@ -107,9 +104,9 @@ private void showBrewingStandInfo(IProbeInfo probeInfo, World world, IProbeHitDa
                 int fuel = ((TileEntityBrewingStand) te).getField(1);
                 probeInfo.horizontal(probeInfo.defaultLayoutStyle().alignment(ElementAlignment.ALIGN_CENTER))
                         .item(new ItemStack(Items.BLAZE_POWDER), probeInfo.defaultItemStyle().width(16).height(16))
-                        .text(LABEL + "Fuel: " + INFO + fuel);
+                        .text(I18n.format("gui.theoneprobe.default_provider.brewing.fuel", LABEL, INFO, fuel));
```
Comment (Snownee) [inline src/main/java/mcjty/theoneprobe/apiimpl/providers/DefaultProbeInfoProvider.java:107]: Suggestion: do not put formatting texts into language file. It is fixed

File: src/main/java/mcjty/theoneprobe/apiimpl/providers/HarvestInfoTools.java [hunk 3]
```diff
@@ -53,7 +55,9 @@ static void showHarvestLevel(IProbeInfo probeInfo, IBlockState blockState, Block
             } else {
                 harvestName = harvestLevels[harvestLevel];
             }
-            probeInfo.text(LABEL + "Tool: " + INFO + harvestTool + " (level " + harvestName + ")");
+            probeInfo.text(I18n.format("gui.theoneprobe.harvest_info_tools.show_harvest_level", LABEL, INFO,
+                    I18n.format("gui.theoneprobe.harvest_info_tools.harvest_tool:" + harvestTool.toLowerCase()),
+                    I18n.format("gui.theoneprobe.harvest_info_tools.harvest_name:" + harvestName.toLowerCase())));
```
Comment (Snownee) [inline src/main/java/mcjty/theoneprobe/apiimpl/providers/HarvestInfoTools.java:60]: I18n for tool classes?

**Comment by liach** [comment]
Hmm, anything blocking this? This looks like a nice pr.

File: src/main/java/mcjty/theoneprobe/apiimpl/providers/DefaultProbeInfoEntityProvider.java [hunk 4]
```diff
@@ -77,10 +74,10 @@ public void addProbeEntityInfo(ProbeMode mode, IProbeInfo probeInfo, EntityPlaye
             }
 
             if (Tools.show(mode, config.getShowMobGrowth()) && entity instanceof EntityAgeable) {
-               int age = ((EntityAgeable) entity).getGrowingAge();
-               if (age < 0) {
-                   probeInfo.text(LABEL + "Growing time: " + ((age * -1) / 20) + "s");
-               }
+                int age = ((EntityAgeable) entity).getGrowingAge();
```
Comment (3TUSK) [inline src/main/java/mcjty/theoneprobe/apiimpl/providers/DefaultProbeInfoEntityProvider.java:77]: Formatting

File: src/main/java/mcjty/theoneprobe/apiimpl/providers/DefaultProbeInfoEntityProvider.java [hunk 5]
```diff
@@ -6,15 +6,12 @@
...
 import mcjty.theoneprobe.apiimpl.styles.LayoutStyle;
 import mcjty.theoneprobe.config.Config;
-import net.minecraft.entity.Entity;
-import net.minecraft.entity.EntityLivingBase;
-import net.minecraft.entity.EntityAgeable;
-import net.minecraft.entity.passive.EntityTameable;
-import net.minecraft.entity.IEntityOwnable;
-import net.minecraft.entity.SharedMonsterAttributes;
+import net.minecraft.client.resources.I18n;
+import net.minecraft.entity.*;
```
Comment (3TUSK) [inline src/main/java/mcjty/theoneprobe/apiimpl/providers/DefaultProbeInfoEntityProvider.java:10]: Guess there shouldn't be star import for consistency...

File: src/main/java/mcjty/theoneprobe/gui/GuiConfig.java [hunk 6]
```diff
@@ -82,31 +82,46 @@ public void drawScreen(int mouseX, int mouseY, float partialTicks) {
...
         for (Preset preset : presets) {
             y = addPreset(x, y, preset);
         }
 
         y += 20;
 
-        RenderHelper.renderText(Minecraft.getMinecraft(), x, y, TextFormatting.GOLD + "Scale:");
+        RenderHelper.renderText(Minecraft.getMinecraft(), x, y, I18n.format("gui.theoneprobe.gui_config.scale.title"));
         y += 12;
-        addButton(x+10, y, 30, 14, "--", () -> { Config.setScale(1.2f);}); x += 36;
```
Comment (3TUSK) [inline src/main/java/mcjty/theoneprobe/gui/GuiConfig.java:105]: Unnecessary formatting changes?

**Review by 3TUSK — CHANGES_REQUESTED** [review]
Some of the `I18n.format` usage may cause dedicate server loading classes that are only available on client side, so thus further testing are needed to ensure no regression are arisen.

**Comment by lihsai0** [comment]
Anyone take over i18n support?
I want to fix the issue @3TUSK mentioned.
However, set up project error for getting `Tesla-1.12.2/1.0.63/Tesla-1.12.2-1.0.63.pom` 404.

**Pull request closed without merging**